### ISSUE TITLE
hcb.size used where size should be

### ISSUE THREAD

**Issue by sterre** [LGSInnovations/sigplot#21]
The attached patch (against a built `sigplot-debug.js`) fixes an apparently erroneous use of `hcb.size` instead of `size`. This is the only instance that my search-fu found, but there may be others that I missed.

[sigplot-debug.js.hcb_size.patch.gz](https://github.com/LGSInnovations/sigplot/files/2202932/sigplot-debug.js.hcb_size.patch.gz)

**Comment by amatma** [comment]
@sterre are there any noticeable bugs/errors you encountered because of this? It is helpful for us to know for testing purposes

**Comment by sterre** [comment]
There was definitely a behavioral problem, but it was about a year ago, in the context of a fairly complicated system, and reported to me by someone else. I don't have a reproducer. However, I do know that the change in question fixed the behavior for the original reporter, so at least HE thought it helped :)

**Comment by sterre** [comment]
Note also that this is issue #27 on your internal Gitlab. I see there are a few more details there.

**Comment by maihde** [comment]
This fix appears to already be included on the `master` branch so I'm going to close this.

**Comment by sterre** [comment]
Is this line correct, then?

https://github.com/LGSInnovations/sigplot/blob/fa39eddb0b491525115f4c63faa29734250322c6/js/sigplot.layer1d.js#L343

**Comment by maihde** [comment]
Yes, that line is correct.  On line 334 there is `this.size = 0` so on line 343 is would be incorrect to use `this.size` within the `reload()` method.

As per the internal ticket 27, the method that was reported broken was push(), and in response line 378 was changed:

https://github.com/LGSInnovations/sigplot/blob/fa39eddb0b491525115f4c63faa29734250322c6/js/sigplot.layer1d.js#L378

**Comment by sterre** [comment]
Thank you for setting my mind at ease :)
